File: frontend/src/types/report.ts
import type { MigrationRecord } from "./migration";
import type { RecommendationResponse } from "./recommendation";

export interface ReportSummary {
  request_id: string;
  recommended_approach: string;
  confidence: string;
  score: number;
  rules_version: string;
  why: string[];
}

export interface RecommendationReport {
  report_id: string;
  request_id: string;
  generated_at: string;
  format: string;
  summary: ReportSummary;
  migration: MigrationRecord;
  recommendation: RecommendationResponse;
}

File: backend/app/schemas/report.py
from __future__ import annotations

from datetime import datetime, timezone

from pydantic import BaseModel, Field

from backend.app.schemas.migration import MigrationRecord
from backend.app.schemas.recommendation import RecommendationResponse


class ReportSummary(BaseModel):
    request_id: str
    recommended_approach: str
    confidence: str
    score: int
    rules_version: str
    why: list[str]


class RecommendationReport(BaseModel):
    report_id: str
    request_id: str
    generated_at: datetime = Field(
        default_factory=lambda: datetime.now(timezone.utc),
    )
    format: str = "json"
    summary: ReportSummary
    migration: MigrationRecord
    recommendation: RecommendationResponse
